codegen/core/src/main/java/software/amazon/smithy/python/codegen/writer/PythonWriter.java

```diff
@@ -141,9 +150,78 @@ public PythonWriter writeDocs(String docs) {
      * @return Formatted documentation.
      */
     public String formatDocs(String docs) {
-        // TODO: write a documentation converter to convert markdown to rst
-        return StringUtils.wrap(docs, CodegenUtils.MAX_PREFERRED_LINE_LENGTH - 8)
-                .replace("$", "$$");
+        String rstDocs = wrapRST(docs).toString();
+        return rstDocs.replace("$", "$$");
+    }
+
+    public static String wrapRST(String text) {
+        StringBuilder wrappedText = new StringBuilder();
+        String[] lines = text.split("\n");
+        for (String line : lines) {
+            wrapLine(line, wrappedText);
+        }
+        return wrappedText.toString();
+    }
+
+    private static void wrapLine(String line, StringBuilder wrappedText) {
+        int indent = getIndentationLevel(line);
+        String indentStr = " ".repeat(indent);
+        line = line.trim();
+
+        while (line.length() > MAX_LINE_LENGTH) {
+            int wrapAt = findWrapPosition(line, MAX_LINE_LENGTH);
+            wrappedText.append(indentStr).append(line, 0, wrapAt).append("\n");
+            line = line.substring(wrapAt).trim();
+            if (line.isEmpty()) {
+                return;
+            }
+        }
+        wrappedText.append(indentStr).append(line).append("\n");
+    }
+
+    private static int findWrapPosition(String line, int maxLineLength) {
+        // Find the last space before maxLineLength
+        int wrapAt = line.lastIndexOf(' ', maxLineLength);
+        if (wrapAt == -1) {
+            // If no space found, force wrap at maxLineLength
+            wrapAt = maxLineLength;
```

Comment:
How does this work for things like links and code that may stretch beyond the limit. Is this going to create rendering issues?

Reply:
It does create some; with the current docs there is only one link I can find with this issue.  

What is the ideal behavior here?  Should we allow these docs to exceed the max line length?

Reply:
If it's a link, probably. That only affects rendering in the code. There's a question of if we even want the links in the docstrings, but I don't think we need to solve that now.